fgspectra/model.py

```diff
@@ -0,0 +1,127 @@
+from abc import ABC, abstractmethod
+import types
+import inspect
+import yaml
+import numpy as np
+
+class Model(ABC):
```

Comment:
This is Class is the key change. It replace `SED`, `PowerSpectrum` and `CrossSpectrum` which were basically identical